### PR TITLE
Feat/open-source-preparation

### PR DESCRIPTION
**Pull request by MehdiActable**

### CODE REVIEW

File: actableai/tasks/base.py [hunk 1]
```diff
@@ -239,18 +233,16 @@ def _func(task_object, *args, **kwargs):
...
+                Args:
+                    task_object:
+                        The AAITask object (self)
+                    args:
+                        The arguments to pass to the function
+                    kwargs:
+                        The named arguments to pass to the function
+
+                Returns:
+                    The result of the function
```
Comment (pckhoi) [inline actableai/tasks/base.py:245]: This looks like Google docstring style to me. I wonder if it displays well in Sphinx.

**Comment by pckhoi** [comment]
Do we use [Black](https://github.com/psf/black)? If not then we should IMHO. That way we never worry about code style again. If we're going to apply Black, then we should format all files at once in a separate PR.

**Comment by pckhoi** [comment]
I don't see any Github workflow to build and release on PyPI based on tag so maybe add such a workflow like [this](https://github.com/pckhoi/deba/blob/main/.github/workflows/taggedrelease.yml).

File: actableai/tasks/base.py (hunk 1 above)
Comment (MehdiActable) [reply actableai/tasks/base.py:245]: This is the format we used for our documentation. Here is how it renders :
https://lib.actable.ai/actableai.tasks.html#module-actableai.tasks.base

**Comment by MehdiActable** [comment]
> Do we use [Black](https://github.com/psf/black)? If not then we should IMHO. That way we never worry about code style again. If we're going to apply Black, then we should format all files at once in a separate PR.

Yes we do and if you go to scripts/pre-push you will see I created a hook that ensures that the modified files are the same when using black and also checks if flake8 is happy with the PEP rules.
I am not very happy with the way its done but I also dont like using black after push on last commit.
Maybe a workflow that checks if the code is black compliant ?

**Comment by MehdiActable** [comment]
>

**Comment by MehdiActable** [comment]
> I don't see any Github workflow to build and release on PyPI so I guess it's manual for now? We should automate that. I have such a workflow triggered on tag push [here](https://github.com/pckhoi/deba/blob/main/.github/workflows/taggedrelease.yml).

I think it would be a very good idea ! The only reason I didnt do it is a lack of time.
Lets try to merge this PR first and create another ticket where we compile those additional features ?

File: actableai/tasks/base.py (hunk 1 above)
Comment (pckhoi) [reply actableai/tasks/base.py:245]: It looks fine so far.

**Comment by pckhoi** [comment]
I don't think black is for compliance-checking. Black is there so that we can mostly forget about Flake8, PEPs, and just code. Also, we never have to argue among ourselves what the code should look like. Just use Black. It just saves time.

I use VSCode and I set Black as the Python formatter. Whenever I save the file, the code is automagically formatted. Nothing complex. I think most other editors support this in some way.

**Comment by MehdiActable** [comment]
> I don't think black is for compliance-checking. Black is there so that we can mostly forget about Flake8, PEPs, and just code. Also, we never have to argue among ourselves what the code should look like. Just use Black. It just saves time.
> 
> I use VSCode and I set Black as the Python formatter. Whenever I save the file, the code is automagically formatted. Nothing complex. I think most other editors support this in some way.

True then lets just try to be more consistent on using it. I will setup my VSCode to format on save and ask the others do it aswell.

**Comment by pckhoi** [comment]
> Lets try to merge this PR first and create another ticket where we compile those additional features ?

Yes I think this PR is big enough as is.

**Review by pckhoi — APPROVED** [review]
LGTM

**Pull request merged**